dbmongo/js/globals.ts

```diff
@@ -20,80 +20,148 @@ type CodeNAF = string
 
 type Scope = "etablissement" | "entreprise"
 
-type ReduceIndexFlags = {
-  algo1: boolean
-  algo2: boolean
-}
-
-type BatchValues = { [batchKey: string]: BatchValue }
-
 type CompanyDataValues = {
   key: SiretOrSiren
   scope: Scope
   batch: BatchValues
 }
 
 type CompanyDataValuesWithFlags = CompanyDataValues & {
-  index: ReduceIndexFlags
-}
-
-type BatchValue = {
-  reporder?: { [periode: string]: RepOrder }
-  compact?: { delete: { [dataType: string]: DataHash[] } }
-  effectif?: { [dataHash: string]: Effectif }
-  apconso?: Record<DataHash, ApConso>
-  apdemande?: Record<DataHash, ApDemande>
-  compte?: Record<DataHash, Compte>
-  interim?: Record<Periode, Interim>
-  delai?: Record<DataHash, Delai>
-} & Partial<DonnéesDefaillances> &
-  Partial<DonnéesCotisationsDettes> &
-  Partial<DonnéesCcsf> &
-  Partial<DonnéesSirene> &
-  Partial<DonnéesSireneUL> &
-  Partial<DonnéesEffectifEntreprise> &
-  Partial<DonnéesBdf> &
-  Partial<DonnéesDiane>
-
-type BatchDataType = keyof BatchValue
+  index: {
+    algo1: boolean
+    algo2: boolean
+  }
+}
+
+type BatchValues = Record<BatchKey, BatchValue>
+
+type BatchValue = Partial<
+  DonnéesRepOrder &
+    DonnéesCompact &
+    DonnéesEffectif &
+    DonnéesApConso &
+    DonnéesApDemande &
+    DonnéesCompte &
+    DonnéesInterim &
+    DonnéesDelai &
+    DonnéesDefaillances &
+    DonnéesCotisation &
+    DonnéesDebit &
+    DonnéesCcsf &
+    DonnéesSirene &
+    DonnéesSireneUL &
+    DonnéesEffectifEntreprise &
+    DonnéesBdf &
+    DonnéesDiane
+>
+
+type BatchDataType = keyof BatchValue // => 'reporder' | 'effectif' | 'apconso' | ...
+
+// Définition des types de données
+
+type DonnéesRepOrder = {
+  reporder: Record<Periode, EntréeRepOrder>
+}
+
+type DonnéesCompact = {
+  compact: { delete: { [dataType: string]: DataHash[] } } // TODO: utiliser un type Record<~BatchDataType, DataHash[]>
+}
+
+type DonnéesEffectif = {
+  effectif: Record<DataHash, EntréeEffectif>
+}
+
+type DonnéesApConso = {
+  apconso: Record<DataHash, EntréeApConso>
+}
+
+type DonnéesApDemande = {
+  apdemande: Record<DataHash, EntréeApDemande>
+}
+
+type DonnéesCompte = {
+  compte: Record<DataHash, EntréeCompte>
+}
+
+type DonnéesInterim = {
+  interim: Record<Periode, EntréeInterim>
+}
+
+type DonnéesDelai = {
+  delai: Record<DataHash, EntréeDelai>
+}
+
+type DonnéesDefaillances = {
+  altares: Record<DataHash, EntréeDefaillances>
+  procol: Record<DataHash, EntréeDefaillances>
+}
+
+type DonnéesCotisation = {
+  cotisation: Record<string, EntréeCotisation> // TODO: utiliser un type plus précis que string
```

Comment:
Ce type est les suivants sont des `DataHash`

Reply:
c'est noté, merci ! j'appliquerai ça dans la prochaine PR

Reply:
c'est corrigé dans PR #80.